pkg/cli/cmd_helpers.go

```diff
@@ -46,18 +47,39 @@ func errCmdFunc(err error) func(*cobra.Command, []string) error {
 	}
 }
 
-// runECmdFunc returns a cobra RunE function that runs subcommand and saves the
-// config, which may have been modified by subcommand.
-func runECmdFunc(
-	fs machinery.Filesystem,
-	c *config.Config,
-	subcommand plugin.Subcommand,
-	msg string,
-) func(*cobra.Command, []string) error {
+// preRunECmdFunc returns a cobra PreRunE function that loads the configuration file
+// and injects it into the subcommand
+func preRunECmdFunc(subcmd plugin.Subcommand, cfg store.Store, msg string) func(*cobra.Command, []string) error {
 	return func(*cobra.Command, []string) error {
-		if err := subcommand.Run(fs); err != nil {
+		err := cfg.Load()
+		if os.IsNotExist(err) {
+			return fmt.Errorf("%s: unable to find configuration file, project must be initialized", msg)
+		} else if err != nil {
+			return fmt.Errorf("%s: unable to load configuration file: %w", msg, err)
+		}
```

Comment:
I am only concern here. 
SDK has commands that should work when no PROJECT file is found. So, would it break this behaviour?

Reply:
This is only applied to `create api`, `create webhook` and `edit` subcommands. Kubebuilder also has commands that should work without a project configuration file: `init`, `version`, `completion`, ...

Reply:
Could we please comment on that in the code? It makes it easier for us to understand it in the future and keep the project maintained. I mean, could you please add a comment when/why that is called.